ManagementPacks/SquaredUp.EAM.Library/Modules/Scripts/SquaredUp.EAM.Discovery.ps1

```diff
@@ -39,26 +62,33 @@ $discoveryData = $SCRIPT:momapi.CreateDiscoveryData(0, $sourceId, $managedEntity
 $discoveries = $DiscoveriesJson | ConvertFrom-Json
 $objInstancesByInstanceId = @{}
 
+# Get name of the EA we are running against
+#=================================================================================
+$eaId = "unknown"
+$eaDisplayName = "unknown"
+try {
+	$map = $discoveries.NewObjects | Where-Object {$_.TypeId -match "\[Name='EA_[0-9a-f]{32}_Availability'\]"}
+	$app = $map.properties | Where-Object {$_.propertyId -match "\[Name='SquaredUpEAMLibrary!SquaredUp.EAM.Library.Class.Availability'\]/ApplicationId"}
```

Comment:
The script parameter `$managedEntityId` is already the SCOM guid for the EA, so this is unnecessary and you can just use that directly.